### PR TITLE
chore: argocd app create output states whether app exists

### LINKED ISSUE #9969: CLI suggests that it is possible to create the same application twice
From the [Trail of Bits audit](https://github.com/argoproj/argoproj/blob/4324729c1c9d3ea21d22afbf7378921826fd2529/docs/argo_security_final_report.pdf) (item 22): 

> Invoking the same Argo CD CLI command to create an application suggests that the application was created twice, while the second invocation did not create another application (Figure 22.1). This result may be confusing to users who want to create an app but use the same application creation data.
> ```
> $ argocd app create zzzz --repo https://github.com/disconnect3d/z/ --path . --dest-namespace
> default --dest-server https://kubernetes.default.svc --directory-recurse
> application 'zzzz' created
> $ argocd app create zzzz --repo https://github.com/disconnect3d/z/ --path . --dest-namespace
> default --dest-server https://kubernetes.default.svc --directory-recurse
> ```
> Recommendation
> Short term, change the Argo CD logic so the Argo CD CLI errors out if a user attempts to create an application with the same data.

### PR DESCRIPTION
**Pull request by danielhelfand**
Closes #9969 

Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Checklist:

* [ ] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ ] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] Optional. My organization is added to USERS.md.
* [ ] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: server/application/application.go [hunk 1]
```diff
@@ -211,7 +211,7 @@ func (s *Server) Create(ctx context.Context, q *application.ApplicationCreateReq
 		reflect.DeepEqual(existing.Finalizers, a.Finalizers)
 
 	if equalSpecs {
-		return existing, nil
+		return existing, status.Errorf(codes.AlreadyExists, "application '%s' already exists and no changes needed", q.Application.Name)
```
Comment (crenshaw-dev) [inline server/application/application.go:214]: I'm a bit hesitant to make a breaking API change. Maybe this is a change for 3.0?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]

File: server/application/application.go (hunk 1 above)
Comment (danielhelfand) [reply server/application/application.go:214]: An alternative to make this non-breaking might be to compare the app returned by the server with the app sent in the request from the CLI. I had tried that before, but couldn't quite get it to work. I can look into it again.

Comment (crenshaw-dev) [reply server/application/application.go:214]: That would be nice... I'd consider it a low-priority issue, but if you figure it out, that'd be great!

File: cmd/argocd/commands/app.go [hunk 2]
```diff
@@ -162,11 +162,16 @@ func NewApplicationCreateCommand(clientOpts *argocdclient.ClientOptions) *cobra.
...
 				}
+
 				created, err := appIf.Create(ctx, &appCreateRequest)
 				errors.CheckError(err)
-				fmt.Printf("application '%s' created\n", created.ObjectMeta.Name)
-			}
 
+				if hasAppChanged(appCreateRequest.Application, created, upsert) {
+					fmt.Printf("application '%s' unchanged\n", created.ObjectMeta.Name)
+				} else {
```
Comment (danielhelfand) [inline cmd/argocd/commands/app.go:171]: Could also have a case for upsert==true where we say application '%s' updated if we want as well as opposed to always saying created.

File: server/application/application.go (hunk 1 above)
Comment (danielhelfand) [reply server/application/application.go]: Added in some checking logic to the CLI so the server doesn't return an error.

File: cmd/argocd/commands/app.go (hunk 2 above)
Comment (crenshaw-dev) [reply cmd/argocd/commands/app.go:171]: I like that. Want to do it in this PR or let that be later?

**Comment by crenshaw-dev** [comment]
Merging master will fix the readthedocs failure.

File: cmd/argocd/commands/app.go (hunk 2 above)
Comment (danielhelfand) [reply cmd/argocd/commands/app.go:181]: Added it in.

**Review by crenshaw-dev — APPROVED** [review]
Brilliant, thanks @danielhelfand!

**Pull request merged**